components/calendar/src/hebrew.rs

```diff
@@ -0,0 +1,848 @@
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2217
+    #[allow(dead_code)]
+    fn last_month_of_hebrew_year(h_year: i32) -> u8 {
+        if Self::is_leap_year(h_year) {
+            ADARII
+        } else {
+            ADAR
+        }
+    }
+
+    // Sabbatical Year of the Hebrew Calendar
+    // Lisp code refernence: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2224
+    #[allow(dead_code)]
+    fn hebrew_sabbatical_year(h_year: i32) -> bool {
+        div_rem_euclid(h_year, 7).1 == 0
+    }
+
+    // Number of days elapsed from the (Sunday) noon prior to the epoch of the Hebrew Calendar to the molad of Tishri of Hebrew year (h_year) or one day later
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2261
+    #[allow(dead_code)]
+    fn hebrew_calendar_elapsed_days(h_year: i32) -> i32 {
+        let months_elapsed = libm::floor((1.0 / 19.0) * (235.0 * h_year as f64 - 234.0));
+        let parts_elapsed = 12084.0 + 13753.0 * months_elapsed;
+        let days = 29.0 * months_elapsed + libm::floor(parts_elapsed / 25920.0);
+
+        if div_rem_euclid_f64(3.0 * (days + 1.0), 7.0).1 < 3.0 {
+            days as i32 + 1
+        } else {
+            days as i32
+        }
+    }
+
+    // Delays to start of Hebrew year to keep ordinary year in range 353-356 and leap year in range 383-386
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2301
+    #[allow(dead_code)]
+    fn hebrew_year_length_correction(h_year: i32) -> u8 {
+        let ny0 = Self::hebrew_calendar_elapsed_days(h_year - 1);
+        let ny1 = Self::hebrew_calendar_elapsed_days(h_year);
+        let ny2 = Self::hebrew_calendar_elapsed_days(h_year + 1);
+
+        if (ny2 - ny1) == 356 {
+            2
+        } else if (ny1 - ny0) == 382 {
+            1
+        } else {
+            0
+        }
+    }
+
+    // Fixed date of Hebrew new year
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2294
+    #[allow(dead_code)]
+    pub(crate) fn hebrew_new_year(h_year: i32) -> RataDie {
+        RataDie::new(
+            FIXED_HEBREW_EPOCH.to_i64_date()
+                + Self::hebrew_calendar_elapsed_days(h_year) as i64
+                + Self::hebrew_year_length_correction(h_year) as i64,
+        )
+    }
+
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2315
+    #[allow(dead_code)]
+    fn days_in_hebrew_year(h_year: i32) -> u16 {
+        (Self::hebrew_new_year(1 + h_year) - Self::hebrew_new_year(h_year)) as u16
+    }
+
+    // True if the month Marheshvan is going to be long in given Hebrew year
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2321
+    #[allow(dead_code)]
+    fn long_marheshvan(h_year: i32) -> bool {
+        let coll = [355, 385];
```

Comment:
suggestion: rename more descriptively
```suggestion
        let long_marheshavan_year_lengths = [355, 385];
```

Reply:
done!